app/src/main/java/com/example/ithaca_transit_android_v2/NetworkingAdapters.kt

```diff
@@ -44,4 +46,128 @@ class LocationAdapter {
         }
         return DataLocation(final)
     }
+}
+
+@JsonClass(generateAdapter = true)
+class RouteAdapter {
+    class DataRoute(
+        val data: DataRouteOptions
+    )
+
+    @JsonClass(generateAdapter = true)
+    class DataRouteOptions(
+        val fromStop: List<JsonRoute>,
+        val walking: List<JsonRoute>,
+        val boardingSoon: List<JsonRoute>
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonRoute(
+        @Json(name = "directions")
+        val directions: List<JsonDirection>,
+        @Json(name = "startCoords")
+        val startLocation: JsonLocation,
+        @Json(name = "endCoords")
+        val endLocation: JsonLocation,
+        // TODO: need to calculate isWalkingOnly
+        @Json(name = "arrivalTime")
+        val arrival: String, // e.g. 2019-11-06T17:07:20Z
+        @Json(name = "departureTime")
+        val depart: String
+        // TODO: need to calculate BoardInMin
+    )
```

Comment:
We will probably need to add the variables totalDuration and totalDistance to our Route model, and we should add that here since backend passes us these values.

Reply:
@JiangoJ Looking at the designs, I don't think there's going to be a need for both values. Remember that our models does not need to contain every value that is returned from the backend, only the values that we know we need. This prevents us from cluttering our own models.

Reply:
The designs have tags by the route that displays the total time for trip with buses and trip that is only walking. That means we might need the totalDuration. However, I am not sure if this is a necessary v0 feature.